tests/profiling/collector/test_stack.py

```diff
@@ -89,6 +91,17 @@ def test_collect_once():
         pytest.fail("Unable to find MainThread")
 
 
+@pytest.mark.skipif(sys.version_info >= (3, 11, 0), reason="PyFrameObjects are lazy-created objects in Python 3.11+")
+def test_collect_once_ensure_all_frames_gc():
+    # Regression test for memory leak with lazy PyFrameObjects in Python 3.11+
+    r = recorder.Recorder()
+    s = stack.StackCollector(r)
+    s._init()
+    all_events = s.collect()
+    assert len(all_events) == 2
+    assert all(not isinstance(_, FrameType) for _ in gc.get_objects)
```

Comment:
```suggestion
    assert all(not isinstance(_, FrameType) for _ in gc.get_objects())
```
I'm not sure this test would pass as is. My feeling is that, since we are in a function, there should be live frame objects. We should probably do the following:
- define a local function, e.g. `def foo():pass`
- call `foo` a few times, e.g. `for _ in range(100): foo()`
- check that any objects of type `FrameType` in `gc.get_objects()` are not frames associated to the function `foo` (we might want to call `gc.collect()` as well before the check to be extra sure we don't race with the GC on those frame objects).